sycl/piapi/src/pi.cpp

```diff
@@ -251,29 +247,27 @@ bool bindPlugin(void *Library, PiPlugin *PluginInformation) {
 }
 
 bool trace(TraceLevel Level) {
-  auto TraceLevelMask = SYCLConfig<SYCL_PI_TRACE>::get();
+  auto TraceLevelMask = config::trace_level_mask();
   return (TraceLevelMask & Level) == Level;
 }
 
 // Initializes all available Plugins.
-const vector_class<plugin> &initialize() {
-  static std::once_flag PluginsInitDone;
-  static vector_class<plugin> *Plugins = nullptr;
-
-  std::call_once(PluginsInitDone, []() {
+const std::vector<plugin> &initialize() {
+  static std::vector<plugin> *Plugins = []() {
```

Comment:
Please, take a look at the comments of https://github.com/intel/llvm/pull/1614 justifying the need for `PluginsInitDone` flag.
Tagging @smaslov-intel and @kholland-intel.

Reply:
The main reason I did it was because I was seeing strange segfaults because of `call_once`, somehow related to `pthreads`. I linked in `Threads::Threads`, but that seemed to have only moved the segfault. As far as I can tell, this code is functionally identical: static initialization is guaranteed to be thread safe and to happen only once. Is there something that I'm missing? I'm not sure from the discussion you linked.